test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java

```diff
@@ -669,6 +670,11 @@ public static void assertVersionSerializable(Version version, Streamable streama
                     equalTo(0));
             assertThat("Serialization failed with version [" + version + "] bytes should be equal for streamable [" + streamable + "]",
                     serialize(version, streamable), equalTo(orig));
+        } catch (UnsupportedOperationException ex) {
+            if (ex.equals("can't read named writeable from StreamInput")) {
```

Comment:
I think @jpountz has managed to get the registry sorted here: https://github.com/elastic/elasticsearch/pull/17546/files#diff-e7168ac65fa4c974595f24e243f2cf7eR652

Maybe you can take advantage of that when that PR is merged?